socs/agents/hwp_pcu/agent.py

```diff
@@ -0,0 +1,245 @@
+import argparse
+import time
+
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+from twisted.internet import reactor
+
+import socs.agents.hwp_pcu.drivers.hwp_pcu as pcu
+
+
+class HWPPCUAgent:
+    """Agent to phase compensation improve the CHWP motor efficiency
+
+    Args:
+        agent (ocs.ocs_agent.OCSAgent): Instantiated OCSAgent class for this agent
+        port (str): Path to USB device in '/dev/'
+
+    """
+
+    def __init__(self, agent, port):
+        self.agent: ocs_agent.OCSAgent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.initialized = False
+        self.take_data = False
+        self.port = port
+        self.status = 'off'
+
+        agg_params = {'frame_length': 60}
+        self.agent.register_feed(
+            'hwppcu', record=True, agg_params=agg_params)
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    @ocs_agent.param('force', default=False, type=bool)
+    def init_connection(self, session, params):
+        """init_connection(auto_acquire=False, force=False)
+
+        **Task** - Initialize connection to PCU
+        Controller.
+
+        Parameters:
+            auto_acquire (bool, optional): Default is False. Starts data
+                acquisition after initialization if True.
+            force (bool, optional): Force initialization, even if already
+                initialized. Defaults to False.
+
+        """
+        if self.initialized and not params['force']:
+            self.log.info("Connection already initialized. Returning...")
+            return True, "Connection already initialized"
+
+        with self.lock.acquire_timeout(0, job='init_connection') as acquired:
+            if not acquired:
+                self.log.warn(
+                    'Could not run init_connection because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            try:
+                self.PCU = pcu.PCU(port=self.port)
+                self.log.info('Connected to PCU')
+            except BrokenPipeError:
+                self.log.error('Could not establish connection to PCU')
+                reactor.callFromThread(reactor.stop)
+                return False, 'Unable to connect to PCU'
+        self.status = self.PCU.get_status()
+        self.initialized = True
+
+        # Start 'acq' Process if requested
+        if params['auto_acquire']:
+            self.agent.start('acq')
+
+        return True, 'Connection to PCU established'
+
+    @ocs_agent.param('command', default='off', type=str)
+    def send_command(self, session, params):
+        """send_command(command)
+
+        **Task** - Send commands to the phase compensation unit.
+        off: The compensation phase is zero.
+        on_1:The compensation phase is +120 deg.
```

Comment:
Why don't we change from 'on_1' to '＋120 deg', 'on_2' to '-120 deg'?

Reply:
I think this might be fine as it is because "on_1" does not necessarily mean +120 deg. We intend to include the hwp_pcu in the hwp_supervisor, so more accurate compensated angle can be described in hwp_supervisor.